src/services/weatherService.js

```diff
@@ -0,0 +1,25 @@
+import axios from '../common/axios-config';
+import { BASE_URL } from '../common/constants';
+
+export const getLocationData = async location => {
+  try {
+    const {
+      data: { locations }
+    } = await axios.get(`${BASE_URL}/location/search/${location}`);
+    return [...locations].sort(
+      (a, b) => a.country.localeCompare(b.country) || a.name.localeCompare(b.name)
+    );
+  } catch (error) {
+    // to be handled accordingly elsewhere
+    return Promise.reject(error);
```

Comment:
Great that you added that. This comment will be also a reminder for me to talk about Promises next time.